cg/meta/report/raredisease.py

```diff
@@ -0,0 +1,115 @@
+            sex=sample_metrics.predicted_sex_sex_check,
+        )
+
+    def is_report_accredited(
+        self, samples: list[SampleModel], analysis_metadata: AnalysisModel = None
+    ) -> bool:
+        """
+        Return whether the Raredisease delivery report is accredited.
+        This method evaluates the accreditation status of each sample's application.
+        """
+        return all(sample.application.accredited for sample in samples)
+
+    def get_scout_uploaded_files(self, case_id: str) -> ScoutReportFiles:
+        """Return Raredisease files that will be uploaded to Scout."""
+        return ScoutReportFiles(
+            snv_vcf=self.get_scout_uploaded_file_from_hk(
+                case_id=case_id, scout_key=ScoutUploadKey.SNV_VCF
+            ),
+            sv_vcf=self.get_scout_uploaded_file_from_hk(
+                case_id=case_id, scout_key=ScoutUploadKey.SV_VCF
+            ),
+            vcf_str=self.get_scout_uploaded_file_from_hk(
+                case_id=case_id, scout_key=ScoutUploadKey.VCF_STR
+            ),
+            smn_tsv=self.get_scout_uploaded_file_from_hk(
+                case_id=case_id, scout_key=ScoutUploadKey.SMN_TSV
+            ),
+        )
+
+    @staticmethod
+    def get_sample_metadata_required_fields(case: CaseModel) -> dict:
+        """Return sample metadata required fields associated to a specific sample."""
+        required_sample_metadata_fields = {}
+        for sample in case.samples:
+            required_fields = (
+                REQUIRED_SAMPLE_METADATA_RAREDISEASE_WGS_FIELDS
+                if PrepCategory.WHOLE_GENOME_SEQUENCING.value
+                in sample.application.prep_category.lower()
+                else REQUIRED_SAMPLE_METADATA_RAREDISEASE_FIELDS
+            )
+            required_sample_metadata_fields.update({sample.id: required_fields})
+        return required_sample_metadata_fields
+
+    def get_required_fields(self, case: CaseModel) -> dict:
+        """Return dictionary with the delivery report required fields for Raredisease."""
+        return {
+            "report": REQUIRED_REPORT_FIELDS,
+            "customer": REQUIRED_CUSTOMER_FIELDS,
+            "case": REQUIRED_CASE_FIELDS,
+            "applications": self.get_application_required_fields(
+                case=case, required_fields=REQUIRED_APPLICATION_FIELDS
+            ),
+            "data_analysis": REQUIRED_DATA_ANALYSIS_RAREDISEASE_FIELDS,
+            "samples": self.get_sample_required_fields(
+                case=case, required_fields=REQUIRED_SAMPLE_RAREDISEASE_FIELDS
+            ),
+            "methods": self.get_sample_required_fields(
+                case=case, required_fields=REQUIRED_SAMPLE_METHODS_FIELDS
+            ),
+            "timestamps": self.get_timestamp_required_fields(
+                case=case, required_fields=REQUIRED_SAMPLE_TIMESTAMP_FIELDS
+            ),
+            "metadata": self.get_sample_metadata_required_fields(case=case),
+        }
```

Comment:
Can we make a pydantic model for this and then return model_to_dict?

RareDiseaseReportRequiredFields(BaseModel) or something

Reply:
Done

changed it for the rest of the workflows as well